### PR TITLE
added ajax branch

### PR DESCRIPTION
**Pull request by rayguna**

### CODE REVIEW

File: app/controllers/comments_controller.rb
Comment (brvarner) [inline app/controllers/comments_controller.rb:1]: Great job with the concise format.js syntax!

File: app/views/comments/_comment.html.erb
Comment (brvarner) [inline app/views/comments/_comment.html.erb:1]: Great use of dom_id and using the remote:true syntax!

File: app/views/follow_requests/_follow_unfollow.html.erb
Comment (brvarner) [inline app/views/follow_requests/_follow_unfollow.html.erb:1]: I'm glad the class selector worked well for you! I added a new div as well.

File: app/views/follow_requests/create.js.erb
Comment (brvarner) [inline app/views/follow_requests/create.js.erb:1]: Love the way you passed in the locals!

File: app/views/photos/_likes.html.erb
Comment (brvarner) [inline app/views/photos/_likes.html.erb:1]: Great job substituting the method: :delete!

File: app/views/likes/destroy.js.erb
Comment (brvarner) [inline app/views/likes/destroy.js.erb:1]: Wow! I'd love to check out the animation here

**Review by brvarner — APPROVED** [review]
Great work Ray!

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -33,6 +34,9 @@ def create
       if @comment.save
         format.html { redirect_back fallback_location: root_path, notice: "Comment was successfully created." }
         format.json { render :show, status: :created, location: @comment }
+
```
Comment (ahuynh3a) [inline app/controllers/comments_controller.rb:37]: Great job linking to the js files. Make sure you are mindful of white spaces and remove any unnecessary ones.

File: app/views/comments/_comment.html.erb [hunk 2]
```diff
@@ -11,12 +11,13 @@
     </div>
     <div>
       <% if current_user == comment.author %>
-        <%= link_to edit_comment_path(comment), class: "btn btn-link btn-sm text-muted" do %>
+        <%= link_to edit_comment_path(comment), class: "btn btn-link btn-sm text-muted", remote: true do %>
```
Comment (ahuynh3a) [inline app/views/comments/_comment.html.erb:14]: Great job being mindful of when to use remote: true and local: false

File: app/views/comments/create.js.erb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+// app/views/comments/create.js.erb
```
Comment (ahuynh3a) [inline app/views/comments/create.js.erb]: remove any unnecessary comments to make your code clean

File: app/views/comments/edit.js.erb [hunk 4]
```diff
@@ -0,0 +1 @@
+$("#<%= dom_id(@comment) %>").replaceWith("<%= j(render "comments/form", comment: @comment) %>");
```
Comment (ahuynh3a) [inline app/views/comments/edit.js.erb]: great job on your comments ray! You can also use variables to make your code more readable.

File: app/views/follow_requests/_follow_unfollow.html.erb [hunk 5]
```diff
@@ -1,17 +1,21 @@
-<% unless sender == recipient %>
-  <% follow_request = current_user.sent_follow_requests.find_by(recipient: recipient) %>
+<div class="<%= dom_id(sender) %>_<%= dom_id(recipient) %>_follow_unfollow">
```
Comment (ahuynh3a) [inline app/views/follow_requests/_follow_unfollow.html.erb:1]: I love that you added a div to capture the unique ids

File: app/views/follow_requests/destroy.js.erb
Comment (ahuynh3a) [inline app/views/follow_requests/destroy.js.erb:1]: good job using variables!

File: app/views/likes/create.js.erb [hunk 6]
```diff
@@ -0,0 +1,15 @@
+var existingLikes = $("#<%= dom_id(@like.photo) %>_likes");
```
Comment (ahuynh3a) [inline app/views/likes/create.js.erb:1]: I love that you added animation to the icons!

**Review by ahuynh3a — APPROVED** [review]
Great job Ray! I think you did a fantastic job. Way to go the extra mile by adding animation to your icon!

File: README.md
Comment (MauritzKurt) [inline README.md:1]: Amazing notes to refer back to!

File: app/views/follow_requests/_follow_unfollow.html.erb (hunk 5 above)
Comment (MauritzKurt) [reply app/views/follow_requests/_follow_unfollow.html.erb:1]: great job utilizing a concise and efficient solution to this problem!

File: app/views/follow_requests/_form.html.erb
Comment (MauritzKurt) [inline app/views/follow_requests/_form.html.erb:1]: Great job keeping consistent formatting!

**Pull request merged**